### PR TITLE
Cell Process panel (and tutorial popups) is on top of menu

### LINKED ISSUE #2326: Cell Process panel (and tutorial popups) is on top of menu
Cell Process panel is on top of menu, making menu buttons unclickable.

To reproduce, open Cell Process menu and drag it to the middle of the screen. Then press escape, you will find the buttons unclickable.

### PR DESCRIPTION
**Pull request by takko**
**Brief Description of What This PR Does**

Created a draggable and resizable Control and changed Cell Process, CheatMenu, EditorCheatMenu and Tutorial Dialogs to this Control.

**Related Issues**

closes #2326

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Comment by RacerBG** [comment]
I had a fair bit of testing with this PR and here are the results:

1. No new crashes or errors - the log is clean.
2. The pop ups which you mentioned are all resizeable and the text formatting seems to adapt pretty well.
3. The cell processes pop up window does not intervene with the menu, the other pop ups are working as intended too.
4. (Optional) There could probably be a maximum resize size for every type of pop up but this might be included in another PR.

In summary: this PR fixes one of my biggest annoyances with the cell processes window and I'm grateful to you for doing this!

**Comment by Kasterisk** [comment]
Regarding the implementation, I'm a bit on the fence about the duplicated logic between the two classes. Just crossed my mind, would it work if we instead make the PauseMenu's root node be a popup? I think that will block all other lower popups.

**Comment by hhyyrylainen** [comment]
I think it would make sense to instead fix the pause menu to be on top of everything else than trying to fix everything else separately.

**Comment by takko** [comment]
But wouldn't that need the Load and Save Dialog to also be on top of the PauseMenu. I think having those TopMost Dialog's is always a problem as you cannot simply control the order.
So either have all (Dialogs, Menus, ...) as Popups or as Controls. Mix is a problem.

I could get rid of some duplicated code via composition. But I did not want to change that code, as it is copied somewhere from godot, and if that code needs change it would be easier to copy the changed godot code again if I do not edit it.

So it's your decision. If you want the PauseMenu as Popup I do that. But I would still keep that new class as it may come handy sometime.

**Comment by hhyyrylainen** [comment]
All of the pause menu content like the load, options, save, would be part of the pause menu popup, so they would stay along with it on top?

**Comment by takko** [comment]
Yes, I guess this may could be. But still, there is also the Extinction Screen Menu. And I'm not sure about Win Menu. Maybe there are more.
I just think that the behavior of the Popup as Dialog is wrong. Not drawn on top but receiving mouse input first is a bug, which we could avoid using Control Dialogs.
I think Popups are only usable as Modal Dialogs.

**Comment by hhyyrylainen** [comment]
Extinction screen and the win screen are contained in the same container, I think.

**Comment by takko** [comment]
They have the same control as holder but are dynamically added via code.
So I guess changing the holder from Control to WindowDialog would be enough.

**Comment by hhyyrylainen** [comment]
Doesn't window dialog have decorations by default? Wasn't the suggestion to use the base popup class to just act as a popup but with no visuals?

**Comment by 84634E1A607A** [comment]
> make the PauseMenu's root node be a popup?

This do solve all the problems.

By the way, resize at corners are not handled in CustomWindowDialog, and I suggest make Dialog resizable zone thicker. The current setting is not that easy to use.

**Comment by 84634E1A607A** [comment]
> So I guess changing the holder from Control to WindowDialog would be enough.

To Popup. Window dialog has backgrounds.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
https://github.com/Revolutionary-Games/Thrive/pull/2877#issuecomment-986000940

**Comment by 84634E1A607A** [comment]
Maybe this can be achieved in another PR.

**Comment by hhyyrylainen** [comment]
If someone wants to make that change as a separate PR, I'll welcome it.

**Comment by 84634E1A607A** [comment]
would it work if we instead make the PauseMenu's root node be a popup? I 

Wouldn't it?

**Comment by hhyyrylainen** [comment]
When I said "If someone wants to make that change as a separate PR, I'll welcome it." I was referring to 

> make the PauseMenu's root node be a popup?

**Comment by 84634E1A607A** [comment]
No. What I wanted to say is that I agree with @Kasterisk , that this pr is actually not worth the trouble.
